freeglut/freeglut/src/x11/fg_main_x11.c

```diff
@@ -579,6 +579,285 @@ __fg_unused static void fghPrintEvent( XEvent *event )
     }
 }
 
+/* UTF-8 decoding routine */
```

Comment:
This should probably get moved up to `fg_misc.c` or `fg_main.c` as it will might be needed for Android and Wayland and isn't X11 specific.